enclave/src/ipfs.rs

```diff
@@ -0,0 +1,74 @@
+use cid::{Cid, Result};
+use ipfs_unixfs::file::adder::FileAdder;
+use multibase::Base;
+use std::convert::TryFrom;
+use std::fs::File;
+use std::io::Read;
+use std::vec::Vec;
+
+pub struct IpfsContent {
+    pub cid: Result<Cid>,
+    pub file_content: Vec<u8>,
+    pub stats: Stats,
+    pub verified: bool,
+}
+
+impl IpfsContent {
+    pub fn new(_cid: Vec<u8>) -> IpfsContent {
+        let cid = std::str::from_utf8(&_cid).unwrap();
+        let mut f = File::open(cid).unwrap();
+        let mut buffer = Vec::new();
+        f.read_to_end(&mut buffer).unwrap();
+
+        println!("reading file {:?} of size {} bytes", f, &buffer.len());
+
+        IpfsContent {
+            cid: Cid::try_from(cid),
+            file_content: buffer,
+            stats: Stats::default(),
+            verified: false,
+        }
+    }
+
+    pub fn verify(&mut self) {
```

Comment:
I am unsure, if we should use the `self.verified` field. Would it be an option to return here  `Resut<(),IPFS_ERROR>` with the error being `IPFS::ERROR::VERIFICATION`? I believe this would be more idiomatic.

This is not a strong opinion though, rather just an input from my side. And if you think differently that is totally fine.

Reply:
I think it depends on how the verification result is used / shared, if verification process will only be a one-off per content then I agree with you.